src/rtapi/rtapi_uspace.hh

```diff
@@ -17,14 +17,21 @@
  */
 #ifndef RTAPI_USPACE_HH
 #define RTAPI_USPACE_HH
+#ifdef __linux__
 #include <sys/fsuid.h>
+#endif
 #include <unistd.h>
 #include <pthread.h>
 
 struct WithRoot
 {
+#ifdef __linux__
     WithRoot() { if(!level) setfsuid(geteuid()); level++; }
     ~WithRoot() { --level; if(!level) setfsuid(getuid()); }
+#else
+    WithRoot() { if(!level) level++; }
```

Comment:
Ok, I suspect I've misunderstood the original intention.  I think it should be something like:

+#else
+    WithRoot() { if(!level) seteuid(0); level++; }

It's still missing any kind of error reporting, though.